Automattic/sensei
# Refactor students page fetching data through Gutenberg and avoiding subqueries

### Changes proposed in this Pull Request

* It removes the subquery from the students query in order to make it work in environments with optimizations where the tables can live in different places.
* It removes an enqueued script that wasn't being used.
* It refactors the courses fetch in order to optimize and simplify using the `useSelect`.
* It introduces a new hook `useSelectWithDebounce` in order to expand the `useSelect`, to work with debounce.
* It removes a middleware we were using to change our rest API arguments to query strings. It seems it's working properly without the middleware with the URL rewrite enabled or not.
* It tweaks the `wp_kses` we use for the Students table, in order to support it in different envs that have customization on the `wp_kses`, like removing the `data-*`.

### Testing instructions


* Create some courses and some students.
* Make some progress with some users.
* Visit WP admin > Sensei LMS > Students, and make sure it's listed properly as previously, especially the _"Last Activity"_ column.
* Add and remove a student from a course, and make sure it works as previously.
